### ISSUE TITLE
IList methods do not throw exceptions for fixed size arrays

### ISSUE THREAD

**Issue by AndreyChechel** [bridgedotnet/Bridge#3584]
Instead of throwing **NotSupportedException**:
- `IList.Insert` does nothing
- `IList.Add` does nothing and returns `-1`
- `IList.Remove` removes an element
- `IList.RemoveAt` throws `TypeError: list.System$Collections$IList$removeAt is not a function at Function.IsIList`
- `IList.Clear` assigns `null` to existing elements

### Steps To Reproduce

https://deck.net/30cb3c32279b4fbe587048a8aa63b975

https://dotnetfiddle.net/AtaSgF

```csharp
public class Program
{
    public static void Main()
    {
        var arr = new int[] { 1, 2, 3 };
        var ilist = (IList)arr;
		
        try
        {
            ilist.Insert(0, 0);
            Console.WriteLine("No Exception");
        }
        catch (Exception ex)
        {
            Console.WriteLine("[Error]: " + ex.Message);
        }
    }
}
```

### Expected Result

Console output:
```
[Error]: Collection was of a fixed size.
```

### Actual Result

Console output:
```
No Exception
```

### See also
- [#3583] IList.IsFixedSize returns incorrect result

**Comment by fabriciomurta** [comment]
After the fix, the fix also applied for #3585 implies in an incorrect exception message for the test case presented here.

**Comment by fabriciomurta** [comment]
Fixed the fix, and the fix for the fix. So, all working. And the correction will make it to the next Bridge release (see milestone).
